### PR TITLE
Adding in support for detecting duplicates of the same version

### PR DESCRIPTION
**Pull request by masterkidan**
Hello, 
This is to address the issue #19 . Essentially this PR adds a flag `ignoreSameVersionDuplicates`, which (if set to `false`) will automatically catch duplicates of the same version of the file when they are added as subdependencies.
By default ignoreSameVersionDuplicates is set to false, this is to prevent failures for the existing consumers of this plugin. 

The change consists of the following updates as well: 

1. Fixed one of the UTs to work for windows file paths. 
2. Added UTs for the newly added functionality
3. It appears that one of the cases under the 'strict' mode was incorrectly written. 
Basically, the test had the following versions of the packages
a -> 1.0.0, 2.0.0, 2.5.0
b -> 1.0.0, 2.0.0
c -> 1.0.0

However the test for strict == false was assuming that there would only be 1 package with a major version change, when there are actually two (as can be seen above).

### CODE REVIEW

**Comment by masterkidan** [comment]
The yarn.lock files updates came when I installed on an ubuntu box, perhaps the original file was checked in on an OSx box?

**Comment by masterkidan** [comment]
Had to explicitly specify node versions in the travis.yml file as 10.0 (latest node) seems to have incompatibilities with the current yarn.lock file 
image

**Comment by alexangas** [comment]
This works great - thanks @masterkidan!

If I had a criticism it would be that it picks up packages in `devDependencies` as well as `dependencies`. I only care about the latter. But perhaps this is an issue with the plugin anyway?

Is this plugin still being maintained @darrenscerri? Is it possible to get this feature merged in?
